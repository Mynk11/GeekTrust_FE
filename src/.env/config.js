export const GET_VEHICLES = "https://findfalcone.herokuapp.com/vehicles";
export const TOKEN_API = "https://findfalcone.herokuapp.com/token";
export const FALCONE_API = "https://findfalcone.herokuapp.com/find";
export const GET_PLANETS = "https://findfalcone.herokuapp.com/planets";
export const NUMBER_OF_PLANETS = 4;