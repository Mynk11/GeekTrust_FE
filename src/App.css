
div.App{
  background-color: rgb(126, 108, 108);
}