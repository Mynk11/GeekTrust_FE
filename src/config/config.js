export const GET_PlANETS = "https://findfalcone.herokuapp.com/planets";
export const GET_VEHICLES = "https://findfalcone.herokuapp.com/vehicles";
export const NUMBER_OF_PLANETS = 4;
export const TOKEN_API = "https://findfalcone.herokuapp.com/token";
export const FALCONE_API = "https://findfalcone.herokuapp.com/find";