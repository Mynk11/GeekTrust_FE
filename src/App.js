import React, { useState } from 'react';
import Footer from './components/footer/footer';
import Header from './components/header/header';
import GetPlanetsDetails from './components/planets/planets';
import SearchButton from './components/button/button';
import usePlanetState from './components/hooks/usePlanetState';
import useVehicleState from './components/hooks/useVehiclHook';
import Result from './components/result/result';
import TextContent from './components/textContent/textContent';
import ErrorBoundary from './components/errorBoundary/errorBondary';
import GetData from './components/getData/getData';
import Token from './components/getData/token';
import FindFalcone from './components/getData/findFalApi';
import { BrowserRouter, Route } from 'react-router-dom';
import './App.css';

function App() {
  const [totalTime, setTotalTime] = useState(0);
  const [distanceObj, setDistanceObj] = useState({});
  const [selectePlanetObj, setSelectePlanetObj] = useState({})
  const [selectedPlanets, setSelectedPlanets] = useState([]);
  const [selectedVehicles, setSelectedVehicles] = useState([]);
  const [planets, setPlanets] = usePlanetState([]);
  const [vehicles, setVehicles] = useVehicleState([]);
  const [result, setResult] = useState({});
  const [time, setTime] = useState({});
  const [selVehicles, setselVehicles] = useVehicleState({});
  const [token, setToken] = useState();
  const [link, setLink] = useState();


  /* Method to reset the state */
  const setAllNull = () => {
    setTotalTime(0);
    setSelectedPlanets([])
    setSelectedVehicles([])
    setDistanceObj({});
    setTime({});
    setselVehicles({});
    setSelectePlanetObj({});
    setResult({});

  }


  return (
    <div className="App container-fluid">
      {/* Component for fetching data calling plants and vehicles api*/}
      <GetData setVehicles={setVehicles} setPlanets={setPlanets} ></GetData>


      <BrowserRouter>
        {/* Component for Header */}
        <Header setAllNull={setAllNull}></Header>
        {/* Component for creatin token */}
        {selectedVehicles.length === 4 ? <Token setLink={setLink} vehicles={vehicles} setToken={setToken}></Token> : ""}
        {/* Component for calling find falcone API */}
        {token ? <FindFalcone token={token} planets={selectedPlanets} vehicles={selectedVehicles} setResult={setResult} /> : ""}
        <Route path="/" exact component={() => {
          return (
            <>
              {/* Component  for static text  */}
              <TextContent props={result}></TextContent>
              <ErrorBoundary>
                {/* Container component for generaing planets dropdown and vehicles option */}
                <GetPlanetsDetails
                  totalTime={totalTime}
                  setTotalTime={setTotalTime}
                  planets={planets}
                  vehicles={vehicles}
                  selectedPlanet={setSelectedPlanets}
                  selectedVehicles={setSelectedVehicles}
                  selectedPlnts={selectedPlanets}
                  selectedVehcle={selectedVehicles}
                  time={time}
                  setTime={setTime}
                  selVehicles={selVehicles}
                  setselVehicles={setselVehicles}
                  selectePlanetObj={selectePlanetObj}
                  setSelectePlanetObj={setSelectePlanetObj}
                  distanceObj={distanceObj}
                  setDistanceObj={setDistanceObj}
                ></GetPlanetsDetails>

                {/* Button component  */}
                <SearchButton link={link} />



              </ErrorBoundary>
            </>
          )
        }}>

        </Route>

        <ErrorBoundary>
          {/* Result continer it has two compnent one for success and other for retry */}
          <Route path="/result" component={() => <Result
            setAllNull={setAllNull}
            time={totalTime}
            result={result}
            setLink={setLink}
          ></Result>}></Route>
        </ErrorBoundary>
        <Footer>
        </Footer>
      </BrowserRouter>
      {/* Static footer component */}
      <Footer>
      </Footer>
    </div>
  );
}

export default App;
