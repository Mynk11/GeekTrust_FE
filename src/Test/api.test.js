import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TimeComponent from '../components/time/time';
import ButtonComponent from '../components/button/button';
import { TOKEN_API, GET_VEHICLES, GET_PLANETS } from '../.env/config';
import App from '../App';




configure({ adapter: new Adapter() });


const mockTime = "230 sec";


describe("API teting", () => {
    it("#Token API Testing", () => {

        fetch(TOKEN_API, {
            "method": "POST",
            "headers": {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            "body": ""
        }).then((suc) => {
            suc.json().then((data) => {
                data.token.expect(String);

            });

        }, err => {
            console.log(`API Fails to feth ${err}`)
        });
    });



    it("#Planets API Testing", () => {

        Promise.all([
            fetch(GET_PLANETS),
            fetch(GET_VEHICLES)
        ]).then((allResponses) => {
            allResponses[0].json().then((planet) => {

                planet.expect([])

            });
            allResponses[1].json().then((vehicle) => {
                vehicle.expect([])

            });

        }).catch((err) => {
            console.log(`API Fails to fetch ${err}`);
        });
    });



    it("#App Mounted Succesfully", () => {
        expect(<App />).toMatchSnapshot();
    });


    it("#Timer Component Mounted Successfully", () => {
        expect(shallow(<TimeComponent time={mockTime}></TimeComponent>)).toMatchSnapshot();
    })

    it("#Button Component Mounted Successfully", () => {
        expect(shallow(<ButtonComponent link={"/"}></ButtonComponent>).length).toEqual(1);
    });


    it('renders an Button', () => {
        var container = shallow(<ButtonComponent link={"/"} />);

        //console.log("===========>", container.debug());
        expect(container.debug()).toMatchSnapshot();;
    });

});



