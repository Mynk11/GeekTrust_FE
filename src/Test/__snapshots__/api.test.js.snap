// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API teting #App Mounted Succesfully 1`] = `<App />`;

exports[`API teting #Timer Component Mounted Successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <timer
    time="230 sec"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        <b>
          TimeTaken   : 
        </b>
         sec
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <b>
            TimeTaken   : 
          </b>,
          undefined,
          " sec",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TimeTaken   : ",
          },
          "ref": null,
          "rendered": "TimeTaken   : ",
          "type": "b",
        },
        undefined,
        " sec",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          <b>
            TimeTaken   : 
          </b>
           sec
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <b>
              TimeTaken   : 
            </b>,
            undefined,
            " sec",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "TimeTaken   : ",
            },
            "ref": null,
            "rendered": "TimeTaken   : ",
            "type": "b",
          },
          undefined,
          " sec",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`API teting renders an Button 1`] = `
"<div className=\\"row text-center btnMargin\\">
  <div className=\\"col-sm-12 text-center\\">
    <Link onClick={[undefined]} className=\\"btn btn-submit  font-weight-bolder color\\" to=\\"/\\" style={{...}}>
      Find falcone!
    </Link>
  </div>
</div>"
`;
